### PR TITLE
Google Tag Manager Implementation

### PR DESCRIPTION
**Pull request by coenraadf**
- add the gatsby-plugin-google-tagmanager and configure it as per the documentation
- add data attributes to all call-to-action buttons on the site.
- update the external link button to accept 2 new params in order to allow for the cta data-attributes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,6 +61,7 @@
     "gatsby-image": "2.2.38",
     "gatsby-plugin-emotion": "4.1.21",
     "gatsby-plugin-google-analytics": "2.1.33",
+    "gatsby-plugin-google-tagmanager": "^2.3.11",
```
Comment (gitdubz) [inline package.json:64]: So for Flamelink App / API and Website we tend to keep the versions fixed and let dependabot help us update for security vulnerabilities etc. 
Can you please change this to be 
"gatsby-plugin-google-tagmanager": "2.3.11"

easiest it to uninstall gatsby-plugin-google-tagmanager and then yarn install gatsby-plugin-google-tagmanager@2.3.11

File: gatsby-config.js [hunk 2]
```diff
@@ -80,6 +80,23 @@ module.exports = {
...
+      ? [
+          {
+            resolve: 'gatsby-plugin-google-tagmanager',
+            options: {
+              id: process.env.GTM_TRACKING_ID,
+              includeInDevelopment: false,
+              routeChangeEventName: 'e_routeChange',
+              defaultDataLayer: {
+                environmentName: process.env.FLAMELINK_ENV,
+                pageType: 'help',
```
Comment (gitdubz) [inline gatsby-config.js:93]: I see Lloyd only added help and inapp, we can make this pageType: page I think

File: gatsby-config.js [hunk 3]
```diff
@@ -80,6 +80,23 @@ module.exports = {
...
     'gatsby-plugin-typescript',
     'gatsby-plugin-emotion',
+    ...(process.env.GTM_TRACKING_ID
+      ? [
+          {
+            resolve: 'gatsby-plugin-google-tagmanager',
+            options: {
+              id: process.env.GTM_TRACKING_ID,
+              includeInDevelopment: false,
+              routeChangeEventName: 'e_routeChange',
```
Comment (gitdubz) [inline gatsby-config.js:90]: nice, I will chat to Lloyd to hear if if this should be e_pageView or what he prefers but we can leave as is for now.

**Review by gitdubz — COMMENTED** [review]
Great stuff!

File: gatsby-config.js (hunk 2 above)
Comment (coenraadf) [reply gatsby-config.js:93]: good idea yes. was wondering about that. will update

File: package.json (hunk 1 above)
Comment (coenraadf) [reply package.json:64]: sure thing. makes sense.

File: gatsby-config.js (hunk 3 above)
Comment (gitdubz) [reply gatsby-config.js:90]: This will probably tell us if the route goes to 404 also
we can check to see if 404 'e_httpError'  event is needed

**Review by gitdubz — APPROVED** [review]

**Pull request merged**